File: app/src/main/java/com/example/subnetapp/adapters/IpArrayAdapter.java
package com.example.subnetapp.adapters;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.TextView;

import com.example.subnetapp.R;

public class IpArrayAdapter extends ArrayAdapter<String> {
  private final Context context;
  private final String[] values;
  private final int[] cidrArr;
  private final int[] hosts;

  public IpArrayAdapter(Context context, String[] values, int[] cidr, int[] hosts) {
    super(context, -1, values);
    this.context = context;
    this.values = values;
    this.cidrArr = cidr;
    this.hosts = hosts;
  }

  @Override
  public View getView(int position, View convertView, ViewGroup parent) {
    LayoutInflater inflater = (LayoutInflater) context
        .getSystemService(Context.LAYOUT_INFLATER_SERVICE);
    View rowView = inflater.inflate(R.layout.ip_list_item, parent, false);

    TextView ipTv = rowView.findViewById(R.id.tvIpItem);
    TextView cidrTv = rowView.findViewById(R.id.cidrIpItem);
    TextView hostsTv = rowView.findViewById(R.id.hostsTv);

    String cidrStr = Integer.toString(cidrArr[position]);
    String hostsStr = "Usable Hosts: " + hosts[position];

    ipTv.setText(values[position]);
    cidrTv.setText("/" + cidrStr);
    hostsTv.setText(hostsStr);

    return rowView;
  }
}

File: app/src/main/java/com/jlrutilities/subnetapp/adapters/IpArrayAdapter.kt
package com.jlrutilities.subnetapp.adapters;

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.TextView
import com.jlrutilities.subnetapp.R


//** Allows Ip information to be passed and reference view information. */
class IpArrayAdapter(
        context: Context,
        values: Array<String>,
        cidr: IntArray,
        hosts: IntArray
): ArrayAdapter<String>(context, -1, values){

  private val context: Context
  private val values: Array<String>
  private val cidrArr: IntArray
  private val hosts: IntArray

  init {
    this.context = context;
    this.values = values;
    this.cidrArr = cidr;
    this.hosts = hosts;
  }


  override fun getView(
          position: Int,
          convertView: View?,
          parent: ViewGroup
  ): View {
    // TODO: check layoutinflator is correct before update to store
    val inflater: LayoutInflater = LayoutInflater.from(context)
    // TODO: ViewHolderPattern
    val rowView: View = inflater.inflate(R.layout.ip_list_item, parent, false)

    val ipTv: TextView = rowView.findViewById(R.id.tvIpItem)
    val cidrTv: TextView = rowView.findViewById(R.id.cidrIpItem)
    val hostsTv: TextView = rowView.findViewById(R.id.hostsTv)

    val cidrStr: String = "/" + cidrArr[position].toString()
    val hostPosition: Int = hosts[position]
    val hostsStr = "Usable Hosts: $hostPosition"

    ipTv.text = values[position]
    cidrTv.text = cidrStr
    hostsTv.text = hostsStr

    return rowView
  }
}
